src/mrinufft/trajectories/io.py

```diff
@@ -0,0 +1,377 @@
+                + "\n"
+                "Max Value : "
+                + str(np.max(np.abs(slew_rate)))
+            )
+    return gradients, start_positions, slew_rate
+
+
+def create_gradient_file(gradients: np.ndarray, start_positions: np.ndarray, 
+                         grad_filename: str, img_size: Tuple[int, ...], 
+                         FOV: Tuple[float, ...], in_out: bool = True,
+                         min_osf: int = 5, gyromagnetic_constant: float = 42.576e3, 
+                         version: float = 4.2, recon_tag: float = 1.1, 
+                         timestamp: Optional[float] = None, keep_txt_file: bool = False):
+    """Create gradient file from gradients and start positions.
+
+    Parameters
+    ----------
+    gradients : np.ndarray
+        Gradients. Shape (num_shots, num_samples_per_shot, dimension).
+    start_positions : np.ndarray
+        Start positions. Shape (num_shots, dimension).
+    grad_filename : str
+        Gradient filename.
+    img_size : Tuple[int, ...]
+        Image size.
+    FOV : Tuple[float, ...]
+        Field of view.
+    in_out : bool, optional
+        Whether it is In-Out trajectory?, by default True
+    min_osf : int, optional
+        Minimum oversampling factor needed at ADC, by default 5
+    gyromagnetic_constant : float, optional
+        Gyromagnetic Constant, by default 42.576e3
+    version : float, optional
+        Trajectory versioning, by default 4.2
+    recon_tag : float, optional
+        Reconstruction tag for online recon, by default 1.1
+    timestamp : Optional[float], optional
+        Timestamp of trajectory, by default None
+    keep_txt_file : bool, optional
+        Whether to keep the text file used temporarily which holds data pushed to 
+        binary file, by default False
+    
+    """
+    num_shots = gradients.shape[0]
+    num_samples_per_shot = gradients.shape[1]
+    dimension = start_positions.shape[-1]
+    if len(gradients.shape) == 3:
+        gradients = gradients.reshape(-1, gradients.shape[-1])
+    # Convert gradients to mT/m
+    gradients = gradients * 1e3
+    max_grad = np.max(np.abs(gradients))
+    file = open(grad_filename + '.txt', 'w')
+    if version >= 4.1:
+        file.write(str(version) + '\n')
+    # Write the dimension, num_samples_per_shot and num_shots
+    file.write(str(dimension) + '\n')
+    if version >= 4.1:
+        img_size = img_size
+        FOV = FOV
+        if type(img_size) is int:
+            img_size = (img_size,) * dimension
+        if type(FOV) is float:
+            FOV = (FOV,) * dimension
+        for fov in FOV:
+            file.write(str(fov) + '\n')
+        for sz in img_size:
+            file.write(str(sz) + '\n')
+        file.write(str(min_osf) + '\n')
+        file.write(str(gyromagnetic_constant * 1000) + '\n')
+    file.write(str(num_shots) + '\n')
+    file.write(str(num_samples_per_shot) + '\n')
+    if version >= 4.1:
```

Comment:
It might be easier to maintain to have a dedicated function for each versiom, a bit of redundancy could be acceptable in this case.

Reply:
For readability? I might disagree as we don't expect much to change really, its just one or 2 additional stuff in the 10 float spaces. We will end up with so much more redundancy.